src/components/BulletCursorOverlay.tsx

```diff
@@ -1,290 +1,86 @@
  * Any position changes from one Thought to another will be animated within this component.
  */
-function PlaceholderTreeNode({
-  children,
-  isTableCol1 = false,
-  x,
-  y,
-  width,
-}: {
-  children?: React.ReactNode
-  isTableCol1?: boolean
-  x: number
-  y: number
-  width: number
-}) {
-  const outerDivStyle = {
-    left: x,
-    top: y,
-    // Table col1 uses its exact width since cannot extend to the right edge of the screen.
-    // All other thoughts extend to the right edge of the screen. We cannot use width auto as it causes the text to wrap continuously during the counter-indentation animation, which is jarring. Instead, use a fixed width of the available space so that it changes in a stepped fashion as depth changes and the word wrap will not be animated. Use x instead of depth in order to accommodate ancestor tables.
-    // 1em + 10px is an eyeball measurement at font sizes 14 and 18
-    // (Maybe the 10px is from .content padding-left?)
-    width: isTableCol1 ? width : `calc(100% - ${x}px + 1em + 10px)`,
-    textAlign: isTableCol1 ? ('right' as const) : undefined,
-  }
-
-  return (
-    <div
-      aria-label='placeholder-tree-node'
-      className={css({
-        transition: 'left {durations.layoutNodeAnimation} linear,top {durations.layoutNodeAnimation} ease-in-out',
-        ...(isTableCol1
-          ? {
-              position: 'relative',
-              width: 'auto',
-            }
-          : {
-              position: 'absolute',
-              width: '100%',
-            }),
-      })}
-      style={{
-        ...outerDivStyle,
-      }}
-    >
-      {children}
-    </div>
-  )
-}
+function PlaceholderTreeNode({ children }: { children?: React.ReactNode }) {
+  const containerRef = React.useRef<HTMLDivElement>(null)
+  const { treeNode } = cursorStore.useState()
+
+  const [targetSvg, setTargetSvg] = useState<SVGSVGElement | null>(null)
+
+  React.useEffect(() => {
+    if (!treeNode) return
+    if (!containerRef.current) return
+
+    const treenodeElement = treeNode as HTMLElement
+    const container = containerRef.current
+
+    // Copy all attributes from the first element (treeNode) to the container
+    Array.from(treenodeElement.attributes).forEach(attr => {
+      container.setAttribute(attr.name, attr.value)
+    })
+
+    // Clear existing content and copy the nested child from treeNode
+    container.innerHTML = ''
+    if (treenodeElement.firstElementChild) {
+      const nestedChild = treenodeElement.firstElementChild.cloneNode(true) as HTMLElement
+      container.appendChild(nestedChild)
+
+      // Find and hide the thought-annotation element
+      const thoughtAnnotation = nestedChild.querySelector('div[aria-label="thought-annotation"]') as HTMLElement
+      if (thoughtAnnotation) {
+        thoughtAnnotation.style.visibility = 'hidden'
+      }
+
+      const staticThought = nestedChild.querySelector('div[aria-label="static-thought"]') as HTMLElement
```

Comment:
This is one of many tight couplings to the DOM. I'd like to consider how we might be able to reduce the surface area to make this more encapsulated and less fragile.

Reply:
yep we can avoiding this pattern by implementing this comment now https://github.com/cybersemics/em/pull/3211/files#r2319893043